python/cugraph/cugraph/dask/structure/mg_property_graph.py

```diff
@@ -650,16 +650,18 @@ def has_duplicate_edges(cls, df):
         """
         Return True if df has >1 of the same src, dst pair
         """
-        # FIXME: this can be very expensive for large DataFrames
         # empty not supported by dask
         if len(df.columns) == 0:
             return False
 
-        def has_duplicate_dst(df):
-            return df[cls.dst_col_name].nunique() != \
-                df[cls.dst_col_name].size
-
-        return df.groupby(cls.src_col_name).apply(has_duplicate_dst).any()
+        # may need to split out for dask frames
+        # to scale
+        unique_pair_len = len(
+            df[[cls.src_col_name, cls.dst_col_name]].drop_duplicates()
+        )
```

Comment:
Just curious, since I think the runtime performance seems worth it, but is there a tradeoff here in the temporary DataFrame using potentially more memory than the groupby object returned from `groupby()`?

Reply:
Thanks for  pointing out the problem around the memory usage . We indeed seem to use more memory here as we now store `unique dst and src combinations` while previously we were only saving  `unique src`  columns.  

That said, with dask i am not too worried as we will release it for each partition as we create it so it should not be too much. 

Also previous to this review we were saving index too, now with `ignore_index=True` we wont save that so that should save on some memory.  Thanks for that.